### PR TITLE
Adding -x option which allows user to specify a threshold and check if atleast those many hosts are active

### PR DESCRIPTION
**Pull request by deepak0004**
The feature as asked by #99

### CODE REVIEW

**Comment by deepak0004** [comment]
Hi @schweikert, 
Could you review this PR?

File: src/fping.c [hunk 1]
```diff
@@ -1323,6 +1332,15 @@ void finish()
     if (stats_flag)
         print_global_stats();
 
+    if (reachable_flag) {
+        if ((num_hosts-num_unreachable >= reachable)) {
+            printf(">=%d hosts are reachable\n", reachable);
+        } else {
+            printf("<%d hosts are reachable\n", reachable);
+        }
+        exit(0);
```
Comment (schweikert) [inline src/fping.c]: I think that for this to be useful it should return a different exit code, depending on failure or success (and hopefully scripts using this won't rely on the output...)

File: src/fping.c [hunk 2]
```diff
@@ -1323,6 +1332,15 @@ void finish()
     if (stats_flag)
         print_global_stats();
 
+    if (reachable_flag) {
+        if ((num_hosts-num_unreachable >= reachable)) {
+            printf(">=%d hosts are reachable\n", reachable);
```
Comment (schweikert) [inline src/fping.c]: Should you maybe also include the number of reachable hosts in the output?

File: src/fping.c (hunk 1 above)
Comment (deepak0004) [reply src/fping.c]: I added different exit codes, 0 for success (if >= N hosts are reachable), and 1 for failure (if < N hosts are reachable).

File: src/fping.c (hunk 2 above)
Comment (deepak0004) [reply src/fping.c]: I have printed the number of reachable hosts instead of the given statement.

File: src/fping.c [hunk 3]
```diff
@@ -1323,6 +1332,16 @@ void finish()
     if (stats_flag)
         print_global_stats();
 
+    if (reachable_flag) {
+        if ((num_hosts-num_unreachable >= reachable)) {
+            printf("Number of reachable hosts: %d\n", num_hosts-num_unreachable);
```
Comment (schweikert) [inline src/fping.c]: I was thinking of maybe something like:

```suggestion
            printf("Enough hosts reachable (required: %d, reachable: %d)\n", reachable, num_hosts-num_unreachable);
```

File: src/fping.c [hunk 4]
```diff
@@ -1323,6 +1332,16 @@ void finish()
...
+            exit(0);
+        } else {
+            printf("<%d hosts are reachable\n", reachable);
```
Comment (schweikert) [inline src/fping.c]: Here too:

```suggestion
            printf("Not enough hosts reachable (required: %d, reachable: %d)\n", reachable, num_hosts-num_unreachable);
```

File: src/fping.c [hunk 5]
```diff
@@ -285,7 +285,7 @@ struct timezone tz;
 /* switches */
 int generate_flag = 0; /* flag for IP list generation */
 int verbose_flag, quiet_flag, stats_flag, unreachable_flag, alive_flag;
-int elapsed_flag, version_flag, count_flag, loop_flag, netdata_flag;
+int elapsed_flag, version_flag, count_flag, loop_flag, netdata_flag, reachable_flag;
```
Comment (schweikert) [inline src/fping.c]: Only one variable should be enough I think (i.e. the variable that contains the value of the reachable flag)

File: src/fping.c [hunk 6]
```diff
@@ -248,7 +248,7 @@ unsigned int interval = DEFAULT_INTERVAL * 100;
 unsigned int perhost_interval = DEFAULT_PERHOST_INTERVAL * 100;
 float backoff = DEFAULT_BACKOFF_FACTOR;
 unsigned int ping_data_size = DEFAULT_PING_DATA_SIZE;
-unsigned int count = 1;
+unsigned int count = 1, reachable = 1;
```
Comment (schweikert) [inline src/fping.c]: I suggest to rename "reachable" to something more descriptive like "min_reachable". Also, the default should be 0?

```suggestion
unsigned int count = 1, min_reachable = 0;
```

File: src/fping.c (hunk 3 above)
Comment (deepak0004) [reply src/fping.c]: Made the required modification in both test cases and the code.

File: src/fping.c (hunk 4 above)
Comment (deepak0004) [reply src/fping.c]: Made the required modification in both test cases and the code.

File: src/fping.c (hunk 5 above)
Comment (deepak0004) [reply src/fping.c]: Removed the reachable_flag variable (taking min_reachable>0 value as the flag as proposed).

File: src/fping.c (hunk 6 above)
Comment (deepak0004) [reply src/fping.c]: Changed the initialisation.

**Comment by schweikert** [comment]
Thanks! I will revert the option to be called --reachable again instead of --min_reachable, because I think it looks nicer.

**Pull request merged**